src/Networking/NexusMods.Networking.NexusWebApi/NexusModsLibrary.cs

```diff
@@ -147,4 +206,40 @@ public async Task<IJobTask<NexusModsDownloadJob, AbsolutePath>> CreateDownloadJo
 
         return nexusJob;
     }
+    
+    public async Task<IJobTask<NexusModsDownloadJob, AbsolutePath>> CreateDownloadJob(
```

Comment:
Implementation of this method can be deduplicated with the other overload.

The difference here and the above overload is that the parameters are extracted from NXMModUrl, while here they are explicit. Aside from that, implementation doesn't differ. The above overload should be able to simply call this overload.

However to make calling more hassle free for public API users, since you don't want `nxmData` as additional parameter make a private method like `CreateDownloadJobImpl` and call that from both methods.